src/Utils.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.16;
+import { BadgeStorage } from "./BadgeStorage.sol";
+import { SignatureCheckerUpgradeable } from "@openzeppelin-upgradeable/utils/cryptography/SignatureCheckerUpgradeable.sol";
+import { ISpecDataHolder } from "./interfaces/ISpecDataHolder.sol";
+import { IERC721Metadata } from "./interfaces/IERC721Metadata.sol";
+
+bytes32 constant AGREEMENT_HASH = keccak256("Agreement(address active,address passive,string tokenURI)");
+
+contract Utils is IERC721Metadata, BadgeStorage {
+  function getBadgeIdHash(address _to, string memory _uri) public view virtual returns (bytes32) {
```

Comment:
Why does this belong here in Utils?

Reply:
My thought was that it felt like some kind of helper function but I guess it's pretty specific to the minting function and not generalizable